packages/patternfly-4/react-charts/src/components/ChartUtils/chart-legend.ts

```diff
@@ -27,56 +39,92 @@ export const getChartPadding = (chartType: string = 'chart', theme: ChartThemeDe
+
+// Returns legend dimensions
+export const getLegendDimensions = ({
+  legendData,
+  legendOrientation,
+  legendProps,
+  theme
+}: ChartLegendDimensionsInterface) => {
+  if (legendData || legendProps.data) {
+    return (VictoryLegend as any).getDimensions({
+      data: legendData,
+      orientation: legendOrientation,
+      theme,
+      ...legendProps // override above
+    });
+  }
+  return {};
+}
 
 // Returns x coordinate for legend
 export const getLegendX = ({
-  chartOrientation,
+  chartWidth,
+  dx = 0,
+  legendData,
+  legendPosition,
   legendOrientation,
   legendWidth,
-  theme,
-  width
+  svgWidth
 }: ChartLegendPaddingXInterface) => {
   if (!legendWidth) {
     return 0;
   }
-  const offset = legendOrientation === 'vertical' ? 22 : 4;
-  const gutter = legendOrientation === 'vertical' ? theme.legend.gutter : 0;
-  const padding = gutter + offset;
-  switch (chartOrientation) {
-    case 'left':
-      return width > legendWidth ? width - legendWidth : 0;
-    case 'top':
-      return width > legendWidth - padding ? Math.round((width - (legendWidth - padding)) / 2) : 0;
+  const borderPadding = getBorderPaddingWorkAround(legendData, legendOrientation);
+
+  const test = Math.round((svgWidth - (legendWidth - borderPadding)) / 2) + dx;
+
+  switch (legendPosition) {
+    case 'bottom':
+      return svgWidth > legendWidth - borderPadding
+        ? Math.round((svgWidth - (legendWidth - borderPadding)) / 2) + dx : dx;
+    case 'right':
+      return chartWidth + CommonStyles.legend.margin + dx;
     default:
-      return 0;
+      return dx;
   }
 };
 
 // Returns y coordinate for legend
 export const getLegendY = ({
-  chartDy = 0,
   chartHeight,
-  chartOrientation,
+  dy = 0,
+  legendPosition,
   chartType,
-  height,
   legendData,
   legendHeight,
   theme
 }: ChartLegendPaddingYInterface) => {
-  // Todo: Get padding for other types of charts
+  const getLegendPadding = (legendData: any[]) => (legendData && legendData.length > 0 ? 15 : 0);
   const dHeight = chartHeight ? chartHeight + getChartPadding(chartType, theme) : 0;
   const lHeight = legendHeight ? legendHeight + getLegendPadding(legendData) : 0;
 
-  switch (chartOrientation) {
+  switch (legendPosition) {
+    case 'bottom':
+      return chartHeight + CommonStyles.legend.margin + dy;
     case 'left':
-      return dHeight > lHeight ? Math.round((dHeight - lHeight) / 2) + chartDy : 0;
+      return dHeight > lHeight ? Math.round((dHeight - lHeight) / 2) + dy : dy;
     case 'right':
-      return dHeight > lHeight ? Math.round((dHeight - lHeight) / 2) + chartDy : 0;
-    case 'top':
-      return height > lHeight ? height - lHeight + chartDy : 0;
+      return dHeight > lHeight ? Math.round((dHeight - lHeight) / 2) + dy : dy;
     default:
-      return chartDy;
+      return dy;
+  }
+};
+
+const getBorderPaddingWorkAround = (legendData: any[], legendOrientation: string) => {
```

Comment:
There is a util function in Victory for estimating text width in pixels that might be useful here:
```js
TextSize.approximateTextSize(string, styleObj);
```

Reply:
Thanks, I'll look at that.

Reply:
I used TextSize to help calculate the difference between the actual text size and legend width.